### PR TITLE
[Form] Correctly round model with PercentType and add a rounding_mode option

### LINKED ISSUE #35296: Form "PercentType" parameter "scale" inconsistency with "NumberType"
**Description**

Parameter `scale` of `PercentType` is a bit confusing when you know how it works for the `NumberType` one:

- `NumberType` will round the submitted value, meaning the model will be rounded too; What's saved and what's shown in the view is identical.
- `PercentType` will round the view only, but will save the model with all the decimals sent.

**Example**

Consider this form:

```php
$form = $this
    ->createFormBuilder()
    ->add('number', NumberType::class, ['scale' => 2])
    ->add('percent', PercentType::class, ['type' => 'integer', 'scale' => 2])
    ->getForm();
```

Then submit values with more decimals than the scale is set to and dump the model:

image

Something like:

```php
$form->submit(['number' => 1.234, 'percent' => 1.234]);
dump($form->getData());
```

The `NumberType` will actually round the model. The `PercentType` one will ignore this scale and keep every decimals in the model.

image

But the view will then show:

image

Leading to something strange, in my opinion.

Having the same behaviour for both would be great, or changing the name of this parameter in `PercentType` to make this difference clearer like `view_scale` for example.

Thanks!

### PR DESCRIPTION
**Pull request by VincentLanglet**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #35296
| License       | MIT
| Doc PR        | symfony/symfony-docs#13138

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php [hunk 1]
```diff
@@ -52,12 +103,17 @@ public function __construct(int $scale = null, string $type = null)
...
+            $roundingMode = self::ROUND_HALF_UP;
+        }
+
         if (!\in_array($type, self::$types, true)) {
             throw new UnexpectedTypeException($type, implode('", "', self::$types));
         }
 
         $this->type = $type;
         $this->scale = $scale;
+        $this->roundingMode = $roundingMode;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php]: `$this->roundingMode = $roundingMode ?? self::ROUND_HALF_UP;` would be enough

**Comment by VincentLanglet** [comment]
@nicolas-grekas I made the PR on master because I added an option `rounding_mode`.

But I kinda consider as a bug the fact the value is not rounded in `3.4`, `4.4` and `5.0` versions. Should I make a bugfix PR on theses versions with only the round function but no `rounding_mode` parameter ?

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php]: I made the code consistent with the type, scale and others Transformers. Isn't better ?

**Review by OskarStark — APPROVED** [review]

**Comment by VincentLanglet** [comment]
If the PR https://github.com/symfony/symfony/pull/35781 is accepted, I'll need to include the same fix.

**Comment by VincentLanglet** [comment]
@nicolas-grekas @xabbuh Hi ! Is something missing with this PR ? :)

**Review by OskarStark — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @VincentLanglet.

**Pull request merged**